### PR TITLE
feat: Workflow title/description in workflow list view. Fixes #6529

### LINKED ISSUE #6529: Show workflow title/description in UI
# Summary

Support showing a title and/or description for workflows in the UI, instead of the less readable `metadata.name`.

# Motivation

The `metadata.name` is an identiifer, not a title. It ends up being something like:

> test-someproject-pr-123-aff39ee

It would be much more readable if you could have something like:

> **Build and test**
> `SuperDuperProject` PR #6529: Implement frobbing (aff39ee)

Should be easy enough to read a standard annotation such as `workflows.argoproj.io/description` for this. It'd be nice to support Markdown here, for markup and linking.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by jmeridth**
Screen Shot 2022-10-16 at 10 52 08 PM

Fixes #6529 (partially, no markdown)

- [x] title and description in list view

TL;DR
Add the following to your workflow annotations to get a title/description to show on the workflow list page:

```yaml
spec:
  annotations:
    workflows.argoproj.io/title: 'this is a title'
    workflows.argoproj.io/description: 'this is a description'

```

Signed-off-by: jmeridth <jmeridth@gmail.com>

Fixes #TODO

Please do not open a pull request until you have checked ALL of these:

* [x] Create the PR as draft .
* [x] Run `make pre-commit -B` to fix codegen and lint problems.
* [x] Sign-off your commits (otherwise the DCO check will fail).
* [x] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [x] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [x] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [x] Github checks are green.
* [x] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

**Comment by alexec** [comment]
Maybe no markdown in v1?

**Comment by jmeridth** [comment]
@alexec that sounds good to me.  Will rollback the Markdown.  Cheers for the suggestion.

**Comment by jmeridth** [comment]
Have an issue where, when the list refreshes the first time (based on timer?) it shows the previous `workflow.metadata.name` until the screen is manually refreshed.  The refresh is not including the annotations I believe.  Looking where this might be missing.

**Review by terrytangyuan — COMMENTED** [review]
Are you still working on this?

**Comment by jmeridth** [comment]
@terrytangyuan yes. Will wrap it up in the next 48 hours. Sorry for delay.

**Comment by jmeridth** [comment]
One last hurdle this weekend.  Should have the PR ready for review in the next 24 hours.

**Comment by jmeridth** [comment]
Found the issue with the refresh.  Fixed.  Waiting on CI and then will mark as ready for review.

**Comment by jmeridth** [comment]
@terrytangyuan could you please rerun the cancelled GitHub Action (job)?  It timed out after 8mins.  I don't think my change would cause that.  If that can finish/pass, I'm ready for this change to be reviewed.  Cheers.

**Review by terrytangyuan — APPROVED** [review]
Thank you!
